### PR TITLE
Update spinner component across Jenkins

### PR DESCRIPTION
**Pull request by janfaracik**
# Videos

https://user-images.githubusercontent.com/43062514/139753622-02ecf008-8bc6-4284-887f-3a7665844535.mov

https://user-images.githubusercontent.com/43062514/139753626-f5097998-fa86-4954-9ac6-d8c0d40b877e.mov

### What's new

I've updated the spinners across Jenkins to match the spinner used in the new search input component, the new spinner has the benefit of scaling to retina displays and working in light/dark/custom themes.

Also updates the full screen loader for settings, the loader for dropdowns and the loader for editing descriptions.

### Proposed changelog entries

* Update spinner component across Jenkins

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/resources/lib/layout/spinner.jelly
Comment (timja) [inline core/src/main/resources/lib/layout/spinner.jelly:34]: I don’t think you need this if check the jelly variable will be empty if unset, but does no harm

**Review by timja — COMMENTED** [review]
The spinner feels a bit fast to me, looks weird next to the build status spinning at a much slower rate anyone else feel this? seems really good otherwise

https://user-images.githubusercontent.com/21194782/139805598-91c71ff1-b692-49ed-b5a7-c8dc0cf5f6df.mov

FTR pipeline will need a separate PR to maintain consistency, could you file a PR there to please? Once this is merged we can look at getting it released, ideally it will be compatible across Jenkins versions in some way (maybe a version check)

File: war/src/main/less/base/style.less
Comment (fqueiruga) [inline war/src/main/less/base/style.less:445]: This is a bit problematic IMO, is there no other way? It's a common source of performance issues, and I'd be worried about that, especially if it has an animated item within.

File: war/src/main/less/base/style.less
Comment (fqueiruga) [inline war/src/main/less/base/style.less:434]: I think these should be kept for backwards compatibility. Did you run an search for uses of these classes?

File: war/src/main/less/base/style.less
Comment (fqueiruga) [inline war/src/main/less/base/style.less:462]: lovely!

**Review by fqueiruga — COMMENTED** [review]
Code looks good. I have a few comments for the css. This is a great PR

**Comment by NotMyFault** [comment]
> The spinner feels a bit fast to me, looks weird next to the build status spinning at a much slower rate anyone else feel this? seems really good otherwise
> 
Yeah, it's quite fast. Syncing its speed with the build status spinner may reduce the noise.

**Review by uhafner — COMMENTED** [review]
I like the design! A little bit too fast, though.

File: core/src/main/resources/lib/layout/spinner.jelly
Comment (janfaracik) [reply core/src/main/resources/lib/layout/spinner.jelly:34]: Good spot! Removed :)

File: war/src/main/less/base/style.less
Comment (janfaracik) [reply war/src/main/less/base/style.less:445]: > The spinner feels a bit fast to me, looks weird next to the build status spinning at a much slower rate anyone else feel this? seems really good otherwise
> 
>  new-spinner.mov 
> 
> FTR pipeline will need a separate PR to maintain consistency, could you file a PR there to please? Once this is merged we can look at getting it released, ideally it will be compatible across Jenkins versions in some way (maybe a version check)

I've slowed it down and also added a check for users with [prefers-reduced-motion](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) enabled, in which case they'll get an even slower spinner

**Comment by janfaracik** [comment]
> The spinner feels a bit fast to me, looks weird next to the build status spinning at a much slower rate anyone else feel this? seems really good otherwise
> 
>  new-spinner.mov 
> 
> FTR pipeline will need a separate PR to maintain consistency, could you file a PR there to please? Once this is merged we can look at getting it released, ideally it will be compatible across Jenkins versions in some way (maybe a version check)

I've slowed it down and also added a check for users with [prefers-reduced-motion](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) enabled, in which case they'll get an even slower spinner.

Reduced motion version:
https://user-images.githubusercontent.com/43062514/139838168-45d1e247-6de8-49d5-9f64-390833972a60.mov

File: war/src/main/less/base/style.less
Comment (janfaracik) [reply war/src/main/less/base/style.less:434]: I have, looks like the only use cases of them is where plugins have copied `hudson-behaviour.js` so should be fine :) 

The `.gif` itself has been left in for backwards compat as its still used for `select` input loaders, but should be able to replace that with the future forms PR.

**Review by timja — COMMENTED** [review]
Speed is much better now!

**Review by timja — APPROVED** [review]

**Review by uhafner — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by zbynek — APPROVED** [review]

**Comment by daniel-beck** [comment]
Caused (to a degree) https://issues.jenkins.io/browse/JENKINS-68934

**Comment by janfaracik** [comment]
> Caused (to a degree) https://issues.jenkins.io/browse/JENKINS-68934

Fixed in https://github.com/jenkinsci/jenkins/pull/6687

**Comment by timja** [comment]
you able to quickly resolve conflicts on that?

the stylelint fixes caused a bunch of conflicts 😢

**Comment by janfaracik** [comment]
> you able to quickly resolve conflicts on that?
> 
> the stylelint fixes caused a bunch of conflicts 😢

Sorted now, cheers for updating the PR :)

**Pull request merged**
